CMakeLists.txt

```diff
@@ -3,8 +3,10 @@
 
 cmake_minimum_required(VERSION 3.26)
 include(FetchContent)
-include(cmake/external/onnxruntime_external_deps.cmake)
-
+#include(cmake/external/onnxruntime_external_deps.cmake)
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/extern/vcpkg/scripts/buildsystems/vcpkg.cmake"
```

Comment:
What if a user really needs to supply a toolchain file? For example, if you open the main ORT repo's build.py and search "toolchain", you will see in many cases for cross-compiling we must provide such a file. For example, emsdk has its own toolchain file. Then how to combine them?

Reply:
Given there are a lot of questions that need to be figured out, I suggest making vcpkg optional. It means users can use vcpkg to build and install packages, but it is not mandatory, so that users still can manually fetch and build the dependencies in case vcpkg doesn't support the usage case.  I suggest avoiding mentioning vcpkg in our CMakefiles. Instead, we asks users to:
1. Install vcpkg
2. The install the packages by using "vcpkg install ..."
3. Then our CMake file locate the packages by using the "find_package" function.

What do you think?